src/core/math/blue-noise.js

```diff
@@ -0,0 +1,50 @@
+
+let data = null;
+const initData = () => {
+    if (!data) {
+        const binaryString = atob(base64String);
+
+        // the data stores an array of RGBA 8-bit values, each of the color channels represents
+        // a separate blue noise stream
+        data = Uint8Array.from(binaryString, char => char.charCodeAt(0));
+    }
+};
+
+const blueNoiseData = () => {
+    initData();
+    return data;
+};
+
+/**
+ * Blue noise based random numbers API.
+ *
+ * @ignore
+ */
+class BlueNoise {
+    seed = 0;
+
+    constructor(seed = 0) {
+        this.seed = seed * 4;
+        initData();
+    }
+
+    next() {
+        this.seed = (this.seed + 4) % data.length;
```

Comment:
Strange that next assumes size 4. Perhaps 4 should be a function parameter.

Reply:
The data is RGBA texture format, so basically a 4 grayscale blue noise arrays, stored in the interleaved format.
To get a single stream of blue noise .. we need to sample each 4th value from the array, so the 4 is not something that should be exposed as a parameter - this is just a function of the data available.

Reply:
OIC you're stepping through the 32x32 image and returning values in sequence. Is that still blue noise?

Reply:
This is the best quick answer I could find:

<img width="686" alt="Screenshot 2024-01-15 at 15 18 48" src="https://github.com/playcanvas/engine/assets/11276292/9a5ab6de-93fc-41b0-b53e-f379c90ae1ec">

Reply:
It's likely there are better 1D blue noise sequences we could store as some larger arrays, but this is close enough to give us a better random generator compared to just random values - the important attributes are the lack of clustering and even distribution, without storing additional data.